usercp.php

```diff
@@ -3221,15 +3221,18 @@
 
 			$db->insert_query("joinrequests", $joinrequest);
 
-			foreach($groupleaders[$usergroup['gid']] as $leader)
+			if(in_array($usergroup['gid'], $groupleaders))
```

Comment:
At this point there's no need to verify that on a deeper level - if `$gid` is a valid index, it's an array of `$uid`s: https://github.com/Devilshakerz/mybb/blob/fixes-2206/usercp.php#L3169-L3181